.github/configs/schema-validator/.spectral.yaml

```diff
@@ -1,11 +1,89 @@
-extends: '@ibm-cloud/openapi-ruleset'
-functions: [spellcheckSubstrings]
+extends: 'ibm-oas-wrapper-check-components.js'
+functions: [spellcheckSubstrings,
+            validateXOBFields,
+            refPathExists,
+            validateItemTypeGroupEntries,
+            validateXOBSampleValue,
+            validateItemTypeAgainstElementType]
 rules:
-  major-version-in-path: warn
-  openapi-tags: warn
-  oas3-api-servers: warn
-  oas3-unused-component: warn
+  enum-case-convention: off
+  major-version-in-path: off
+  openapi-tags: off
+  oas3-api-servers: off
+  oas3-unused-component: off
+  property-case-convention: off
+  property-description: off
+  property-inconsistent-name-and-type: off
+  
+  element-valid-item-type:
+    message: 'Item type does not exist: {{error}}'
+    given: $.components.schemas[*].allOf[1][x-ob-item-type]
+    severity: error
+    then:
+      function: validateXOBFields
+      functionOptions:
+        objectPath: x-ob-item-types
+  
+  element-valid-item-type-group:
+    message: 'Item type group does not exist: {{error}}'
+    given: $.components.schemas[*].allOf[1][x-ob-item-type-group]
+    severity: error
+    then:
+      function: validateXOBFields
+      functionOptions:
+        objectPath: x-ob-item-type-groups
 
+  item-type-group-valid-type:
+    message: 'Item type group type does not exist: {{error}}'
+    given: $[x-ob-item-type-groups][*].type
+    severity: error
+    then:
+      function: refPathExists
+  
+  item-type-group-valid-entries:
+    description: The enums or units in the item type group exist in the item type definition.
+    message: '{{error}}'
+    given: $[x-ob-item-type-groups][*]~
+    severity: error
+    then:
+      function: validateItemTypeGroupEntries
+  
+  element-valid-sample-value:
+    description: >-
+      Checks the following items about sample value:
+      - It is an object. The object may be empty (see requireAtLeastOneField).
+      - Its fields are only primitives.
+      - Value has the correct OpenAPI type
+      - If the schema definition OB item type defines enums or units, and it has an item type group:
```

Comment:
Is the check for valid enum or unit skipped if an item type group is not present?

Reply:
The enum or unit sample values are always checked against the item type. If an item type group is present, the enum or unit sample values must also be in the item type group. I updated the validation rule description to describe that.